gitpod-io/gitpod
# [rebuild] auto manage windows

## Description


See #16535 for the motivation and [loom](https://www.loom.com/share/d5186fb9cdf440d3affae419452da372) for demo.

## Related Issue(s)

fix #16535

## How to test


From sources
- Open this RP on gitpod.io
- Run `components/supervisor/validate.sh`
- It would start another workspace and should automatically open a new window.
- Interrupt validate terminal, it should stop the workspace and corresponding windows should be auto closed as well.

## Release Notes

```release-note
NONE
```

## Documentation


## Build Options:

      Experimental feature to run the build with GitHub Actions (and not in Werft).
      leeway-target=components:all
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

</details>

<details>
<summary>Installer Options</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
